### PR TITLE
WELD-2376 Mention validation report in the docs

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: docs/reference/src/main/asciidoc/developmentmode.asciidoc [hunk 1]
```diff
@@ -78,4 +78,27 @@ However, a default UI - HTML client (single-page application) is only available
...
+=====================================
+ Weld - Deployment Validation: FAILED
+ ------------------------------------
+ HTML report generated to:
+
+ file:///path/to/report/weld-validation-report.html
+=====================================
+-----------------------------------------------------------------------------------------------------------
+
+TIP: We encourage you to always attach this report when asking a question on the mailing list or any other communication channel.
```
Comment (manovotn) [inline docs/reference/src/main/asciidoc/developmentmode.asciidoc]: I wish this actually happened...
But who knows, they say hope dies last, right? :-)

File: docs/reference/src/main/asciidoc/developmentmode.asciidoc [hunk 2]
```diff
@@ -78,4 +78,27 @@ However, a default UI - HTML client (single-page application) is only available
...
+TIP: There are some configuration properties which allow to tune or disable Probe features, e.g. to restrict the set of components which will be monitored. See also <<config-dev-mode>>.
+
+
+[[validation-report]]
+==== Validation Report
+
+If a deployment validation fails and the development mode is enabled a simple HTML report is generated.
+The report contains a lot of useful information such as Weld version, list of enabled beans, list of bean archives, Weld configuration, etc.
+By default, the report is generated to the user's current working directory, ie. `user.dir`.
+However, it's also possible to specify a path of the target directory using the `org.jboss.weld.probe.exportDataAfterDeployment` configuration property - see also <<config-dev-mode>>.
```
Comment (manovotn) [inline docs/reference/src/main/asciidoc/developmentmode.asciidoc]: `path of the target directory` -> `path to the target directory`

File: docs/reference/src/main/asciidoc/developmentmode.asciidoc (hunk 1 above)
Comment (mkouba) [inline docs/reference/src/main/asciidoc/developmentmode.asciidoc]: Do not be so pessimistic ;-)

File: docs/reference/src/main/asciidoc/developmentmode.asciidoc (hunk 2 above)
Comment (mkouba) [inline docs/reference/src/main/asciidoc/developmentmode.asciidoc]: Ok, sounds better.

**Pull request merged**